src/factorysetup.c

```diff
@@ -302,6 +329,114 @@ static void _api_msg(const uint8_t* input, size_t in_len, uint8_t* output, size_
     *output_len = out_len;
 }
 
+static void _free(uint8_t** buf)
```

Comment:
I would say that the usual case for this kind of function is that you also set `*buf = NULL` in this function. Was this the intention?

I also think free is defined for NULL as input, so no need to check.

Reply:
Fixed